src/util-privs.c

```diff
@@ -154,7 +155,9 @@ int SCGetUserID(const char *user_name, const char *group_name, uint32_t *uid, ui
 
     /* Get the user ID */
     if (isdigit((unsigned char)user_name[0]) != 0) {
-        userid = atoi(user_name);
+        if (ByteExtractStringUint32(&userid, 10, 0, (const char *)user_name) < 0) {
+            FatalError(SC_ERR_UID_FAILED, "invalid user id value: '%s'", user_name);
```

Comment:
Suggest making other places where `SCLogError` is followed by `exit` into `FatalError` (as a sep commit, of course).

Reply:
That is going to come as a part of https://redmine.openinfosecfoundation.org/issues/3188